### PR TITLE
[SPARK-36846][PYTHON] Inline most of type hint files under pyspark/sql/pandas folder

### PR DESCRIPTION
**Pull request by ueshin**
### What changes were proposed in this pull request?

Inlines type hint files under `pyspark/sql/pandas` folder, except for `pyspark/sql/pandas/functions.pyi` and files under `pyspark/sql/pandas/_typing`.

- Since the file contains a lot of overloads, we should revisit and manage it separately.
- We can't inline files under `pyspark/sql/pandas/_typing` because it includes new syntax for type hints.

### Why are the changes needed?

Currently there are type hint stub files (`*.pyi`) to show the expected types for functions, but we can also take advantage of static type checking within the functions by inlining the type hints.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

### CODE REVIEW

**Review by HyukjinKwon — COMMENTED** [review]
I think it looks fine. @itholic and @xinrong-databricks mind double checking since it needs some human look to verify 😢

File: python/pyspark/sql/pandas/conversion.py [hunk 1]
```diff
@@ -301,27 +314,50 @@ class SparkConversionMixin(object):
...
+    @overload
+    def createDataFrame(
+        self,
+        data: "PandasDataFrameLike",
+        schema: Union[StructType, str],
+        verifySchema: bool = ...,
+    ) -> "DataFrame":
+        ...
+
+    def createDataFrame(  # type: ignore[misc]
```
Comment (xinrong-databricks) [inline python/pyspark/sql/pandas/conversion.py:333]: Since we have `@overload`s, do we have to annotate the original function, here `createDataFrame`?

**Review by xinrong-meng — APPROVED** [review]

**Comment by xinrong-databricks** [comment]
LGTM, thanks!

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**